components/higher-order/with-filters/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import createHooks from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import wrapperDisplayName from '../wrapper-display-name';
+
+// TODO: Use withHooks HOC from https://github.com/WordPress/gutenberg/pull/3321.
+const hooks = createHooks();
+
+export default function withFilters( hookName ) {
```

Comment:
Unit tests would be lovely here, but I prefer to wait for #3321 before I come up with the idea how to test it :)